nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/auth/ColdStorageS3TestHelper.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.coldstorage.helpers.ColdStorageHelper;
+import org.nuxeo.coldstorage.operations.AbstractTestColdStorageOperation;
+import org.nuxeo.ecm.blob.s3.S3BlobProvider;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.BlobManager;
+import org.nuxeo.ecm.core.blob.ManagedBlob;
+import org.nuxeo.runtime.api.Framework;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectListing;
+
+public class ColdStorageS3TestHelper {
+
+    protected static final Logger log = LogManager.getLogger(ColdStorageS3TestHelper.class);
+
+    protected static ColdStorageS3TestHelper INSTANCE;
+
+    protected static final String AWS_MAIN_BUCKET_NAME_ENV_VAR = "COLDSTORAGE_AWS_MAIN_BUCKET_NAME";
+
+    protected static final String AWS_GLACIER_BUCKET_NAME_ENV_VAR = "COLDSTORAGE_AWS_GLACIER_BUCKET_NAME";
+
+    protected static final String AWS_BUCKET_PREFIX_ENV_VAR = "COLDSTORAGE_AWS_BUCKET_PREFIX";
+
+    protected static final String AWS_REGION_ENV_VAR = "COLDSTORAGE_AWS_REGION";
+
+    protected final AWSCredentialsProvider awsCredentialsProvider = new DefaultAWSCredentialsProviderChain();
+
+    protected AWSCredentials credentials;
+
+    protected String mainBucket;
+
+    protected String glacierBucket;
+
+    protected String region;
+
+    protected String bucketPrefix;
+
+    public static ColdStorageS3TestHelper getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ColdStorageS3TestHelper();
+        }
+        return INSTANCE;
+    }
+
+    private ColdStorageS3TestHelper() {
+        this.credentials = awsCredentialsProvider.getCredentials();
+
+        log.info("ColdStorage integration tests will be done using AWS");
+
+        this.mainBucket = Objects.requireNonNull(getEnvValue(AWS_MAIN_BUCKET_NAME_ENV_VAR),
+                "Main bucket name is required");
+        this.glacierBucket = Objects.requireNonNull(getEnvValue(AWS_GLACIER_BUCKET_NAME_ENV_VAR),
+                "Glacier bucket name is required");
+        this.bucketPrefix = Objects.requireNonNull(getEnvValue(AWS_BUCKET_PREFIX_ENV_VAR), "Bucket prefix is required");
+        this.region = Objects.requireNonNull(getEnvValue(AWS_REGION_ENV_VAR), "Region is required");
+    }
+
+    public AWSCredentials getCredentials() {
+        return credentials;
+    }
+
+    public void cleanMainBucket(String blobName) {
+        AmazonS3 amazonS3 = getAmazonS3(blobName);
+        ObjectListing s3Objects = amazonS3.listObjects(this.getMainBucket(), this.getBucketPrefix());
+        s3Objects.getObjectSummaries()
+                 .forEach(s3Object -> amazonS3.deleteObject(this.getMainBucket(), s3Object.getKey()));
```

Comment:
I think you can have a look at [delete-multiple-objects](https://docs.aws.amazon.com/AmazonS3/latest/userguide/delete-multiple-objects.html)

```
   // Delete the sample objects.
            DeleteObjectsRequest multiObjectDeleteRequest = new DeleteObjectsRequest(bucketName)
                    .withKeys(keys)
                    .withQuiet(false);
```